app/services/battle.py

```diff
@@ -0,0 +1,39 @@
+from typing import Dict, List
+
+from app.knights.knight import Knight
+
+
+class Battle:
+    def __init__(self, knights: dict) -> None:
+        self.knights = [Knight(**knights[knight]) for knight in knights]
+
+    def start(self) -> Dict[str, int]:
+        self.knights = self.__sort_knights()
+
+        for i in range(0, len(self.knights) - 1, 2):
+            self.knights[i] = self.__hit(
+                self.knights[i], self.knights[i + 1].power
+            )
+
+            self.knights[i + 1] = self.__hit(
+                self.knights[i + 1], self.knights[i].power
+            )
+
+        return {knight.name: knight.hp for knight in self.knights}
+
+    @staticmethod
+    def __hit(knight: Knight, power: int) -> Knight:
+        knight.hp -= power - knight.protection
+
+        if knight.hp < 0:
+            knight.hp = 0
+        return knight
```

Comment:
You can avoid the overhead of calling a static method for each hit by calculating the damage directly inside the loop.
Also instead of assigning `knight.hp = 0` when it becomes negative, use the `max` function to set it to 0 if it's negative.